msvc/scripts/genproj.cs

```diff
@@ -925,7 +1006,11 @@ public VsCsproj Generate (string library_output, Dictionary<string,MsbuildGenera
 			Replace ("@ADDITIONALLIBPATHS@", String.Empty).
 			Replace ("@RESOURCES@", resources.ToString ()).
 			Replace ("@OPTIMIZE@", Optimize ? "true" : "false").
-			Replace ("@SOURCES@", sources.ToString ());
+			Replace ("@SOURCES@", sources.ToString ()).
+			Replace ("@METADATAVERSION@", assemblyName == "mscorlib" ? "<RuntimeMetadataVersion>Mono</RuntimeMetadataVersion>" : "");
```

Comment:
What does Mono mean there, we should just hardcode the metadata version it's not gonna change

Reply:
As-is we specify no metadata version, so when you start built executables on windows you get a really cryptic error message. One of the CLR developers suggested putting something like 'Mono' in so that the Framework will instead display an error message about Mono. (I lost a couple hours trying to figure out the error when I hit it.)

Reply:
We should specify the metadata version as we do in our makefile, see https://github.com/mono/mono/blob/06b836ed912df491f1000f03d8077d3f36f7336a/mcs/class/corlib/Makefile#L18

Reply:
Hm, that's unfortunate. Do we rely on it having that value? If so, that will make the error message even more confusing on Windows, I think... I'll test it.

Reply:
There should be no error message this is the only reasonable value for .net nowadays anyway

Reply:
There's an error message because we're linked against the mono bcl. The error message isn't going to go away so our option is for it to make no sense or for it to say 'Mono'